### PR TITLE
Add a new default-on transform for 'if' newline consistency

### LINKED ISSUE #52: Enforce newline style in if-expressions
Currently cljfmt allows if-expressions that look like this:

```clojure
(if foo? a
  b)
```

This proposal would have it move the true-branch to its own line when the false-branch is not on the same line as the "if":

```clojure
(if foo?
  a
  b)
```

One-line if-expressions like `(if foo? a b)` would **not** be reformatted.

### PR DESCRIPTION
**Pull request by cespare**
Fixes #52.

### CODE REVIEW

**Comment by cespare** [comment]
(@dmac check out the `caleb.cljfmt-if-consistency` branch if you want to see the effect on our internal codebase)

File: README.md [hunk 1]
```diff
@@ -92,6 +91,23 @@ becomes
 
 Consolidate consecutive blank lines into a single blank line.
 
+### fix-if-newline-consistency (default: on)
+
+Ensure that if one arm of an `if` is one its own line, the other is as well.
```
Comment (dmac) [inline README.md:96]: s/one/on/

File: format/testdata/transform/ifnewlines_before.clj [hunk 2]
```diff
@@ -0,0 +1,5 @@
+(if foo? "a"
+  "b")
+
+(if-let [x 123]
+  456 789)
```
Comment (dmac) [inline format/testdata/transform/ifnewlines_before.clj:5]: Should we include negative test cases? For example, to ensure `(if foo? "a" "b")` stays on one line.

File: format/transform.go [hunk 3]
```diff
@@ -21,45 +21,77 @@ const (
...
 	//     [x] ...)
+	//
 	TransformFixDefmethodDispatchValNewline
 
 	// TransformRemoveExtraBlankLines consolidates consecutive blank lines
 	// into a single blank line.
 	TransformRemoveExtraBlankLines
 
+	// TransformFixIfNewlineConsistency adds a newline before the 'true' arm of
+	// an if expression if the 'false' arm has a newline before it. This:
```
Comment (dmac) [inline format/transform.go:67]: I don't think this description captures this behavior:

```clojure
(if foo?
  "a" "b")

; becomes

(if foo?
  "a"
  "b")
```

Maybe you wrote this before handling that case?

**Review by dmac — APPROVED** [review]

**Pull request merged**